File: src/main/java/me/zax71/lightHub/listeners/PlayerBlockBreak.java
package me.zax71.lightHub.listeners;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import net.minestom.server.entity.Player;
import net.minestom.server.event.EventListener;
import net.minestom.server.event.player.PlayerBlockBreakEvent;
import org.jetbrains.annotations.NotNull;
import org.spongepowered.configurate.ConfigurationNode;

import static me.zax71.lightHub.Main.config;
import static me.zax71.lightHub.Main.configUtils;

/**
 * Stops players from breaking blocks in all worlds
 */
public class PlayerBlockBreak implements EventListener<PlayerBlockBreakEvent> {
    @Override
    public @NotNull Class<PlayerBlockBreakEvent> eventType() {
        return PlayerBlockBreakEvent.class;
    }

    @Override
    public @NotNull Result run(@NotNull PlayerBlockBreakEvent event) {

        final ConfigurationNode protectionErrorNode = config.node("messages", "protectionError");
        Player player = event.getPlayer();

        // https://github.com/Minestom/Minestom/discussions/1596
        event.setCancelled(true);

        // Send an error message
        Component message = MiniMessage.miniMessage().deserialize(
                configUtils.getOrSetDefault(protectionErrorNode, "<bold><red>Hey!</bold> <grey>Sorry, but you can't break that block here")
        );
        player.sendMessage(message);

        return Result.SUCCESS;
    }
}

File: src/main/kotlin/me/zax71/lightHub/listeners/PlayerBlockBreak.kt
package me.zax71.lightHub.listener


import me.zax71.lightHub.config
import me.zax71.lightHub.instances.DefaultInstance
import me.zax71.lightHub.inventories.GameMenu

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage
import net.minestom.server.event.EventListener
import net.minestom.server.event.player.PlayerBlockBreakEvent
import net.minestom.server.instance.Instance

class PlayerBlockBreak : EventListener<PlayerBlockBreakEvent> {

    override fun eventType(): Class<PlayerBlockBreakEvent> {
        return PlayerBlockBreakEvent::class.java
    }

    override fun run(event: PlayerBlockBreakEvent): EventListener.Result {
        val player = event.player

        // https://github.com/Minestom/Minestom/discussions/1596
        event.isCancelled = true
        val message: Component = MiniMessage.miniMessage().deserialize(
            config.getOrSetDefault(
                "protectionError",
                "<bold><red>Hey!</bold> <grey>Sorry, but you can't break that block here"
            )
        )
        player.sendMessage(message)


        player.openInventory(GameMenu.inventory!!)

        return EventListener.Result.SUCCESS
    }
}